src/EventStore.Core/TransactionLog/FileNamingStrategy/VersionedPatternFileNamingStrategy.cs

```diff
@@ -44,6 +45,59 @@ public class VersionedPatternFileNamingStrategy : IFileNamingStrategy {
 			return versions;
 		}
 
+		private int GetFileNumberFor(string path) {
+			var fileName = Path.GetFileName(path);
+			if (!_chunkNamePattern.IsMatch(fileName))
+				throw new ArgumentException($"Invalid file name: {fileName}");
+
+			var start = _prefix.Length;
+			var end = fileName.IndexOf('.', _prefix.Length);
+			Debug.Assert(end != -1);
+
+			if (!int.TryParse(fileName[start..end], out var chunkNumber))
+				throw new ArgumentException($"Invalid file name: {fileName}");
+
+			return chunkNumber;
+		}
+
+		public void EnumerateAllFiles(
+			Func<string, int> getNextFileNumber,
+			Action<string, int, int> onLatestVersionFound = null,
+			Action<string, int> onOldVersionFound = null,
+			Action<string, int> onFileMissing = null) {
```

Comment:
Prefer not to use optional parameters

Reply:
Fixed in 3c1b84b61a4c4db2d03f367d0f74a1fa12106881